src/domain/repositories/UserMonitoringMetadataRepository.ts

```diff
@@ -0,0 +1,68 @@
+import { ProgramMetadata, UserRes, UserRoleAuthority, User } from "data/d2-users/D2Users.types";
+import { Async } from "domain/entities/Async";
+
+export interface UserMonitoringMetadataRepository {
+    getTemplateAuthorities(options: UsersOptions): Promise<Async<TemplateGroupWithAuthorities[]>>;
```

Comment:
[Clean Arc.] Same process, I have identify that you will need a `UserRoleRepository` with methods `getAll` and `getByIds` and UserRepository with a method `getByIds`